packages/server/graphql/queries/tasks.ts

```diff
@@ -1,11 +1,37 @@
-import {GraphQLID, GraphQLInt, GraphQLNonNull, GraphQLBoolean, GraphQLList} from 'graphql'
-import {getUserId, isTeamMember} from '../../utils/authorization'
+import {GraphQLBoolean, GraphQLID, GraphQLInt, GraphQLList, GraphQLNonNull} from 'graphql'
+import isTaskPrivate from 'parabol-client/utils/isTaskPrivate'
+import {getUserId} from '../../utils/authorization'
 import standardError from '../../utils/standardError'
-import {GQLContext} from '../graphql'
+import {DataLoaderWorker, GQLContext} from '../graphql'
 import GraphQLISO8601Type from '../types/GraphQLISO8601Type'
 import {TaskConnection} from '../types/Task'
 import connectionFromTasks from './helpers/connectionFromTasks'
-import isTaskPrivate from 'parabol-client/utils/isTaskPrivate'
+
+const getValidTeamIds = (teamIds: null | string[], tms: string[]) => {
+  // the following comments can be removed pending #4070
+  // const viewerTeamMembers = await dataLoader.get('teamMembersByUserId').load(viewerId)
+  // const viewerTeamIds = viewerTeamMembers.map(({teamId}) => teamId)
+  if (!teamIds) return tms
+  // filter the teamIds array to only teams the user has a team member for
+  return teamIds.filter((teamId) => tms.includes(teamId))
+}
+
+const getValidUserIds = async (
+  userIds: null | string[],
+  viewerId: string,
+  validTeamIds: string[],
+  dataLoader: DataLoaderWorker
+) => {
+  if (!userIds) return null
+  if (userIds.length === 1 && userIds[0] === viewerId) return userIds
+  // NOTE: this will filter out ex-teammembers. if that's a problem, we should use a different dataloader
```

Comment:
this will be a problem if someone wants to see all the tasks that ONLY an ex-team member has done. we can handle that on the next PR